### ISSUE TITLE
[bug]: When using --group_key AND --enable_emission the assets have different tweaked_group_key

### ISSUE THREAD

**Issue by straumat** [lightninglabs/taproot-assets#627]
### Background

When emiting a second emissions of an asset, if you set `--group_key` AND `--enable_emission`, the second asset emited will have a different `tweaked_group_key`.

### Your environment

```
tapd v0.3-rc2
```

### Steps to reproduce
I mint my first asset
```
tapcli assets mint --type normal \ 
                --name unlimitedRoylloCoin1 \
                --enable_emission true
```
I get this result:
```
"name": "unlimitedRoylloCoin1"
"asset_id": "ce2d57a6bef7668a781bc149f5491aab7392b30a2f4368f013722f46a57abc70"
  "asset_group": {
    "raw_group_key": "036c1fb8f682b877b9caa27ddb7068a242f662332ea7eb99e3c32c12170c15a0bc",
    "tweaked_group_key": "0349d60c6689bbbb2ffc9b30b45b96a3d6e5fd5fd01c867344c0054f6105048241",
    "asset_witness": "0140e0c383aaecabb44bee72683d825118b7ef7124a9c8be149c82b7c4c525340aceb4194a177bc1e13e0bee7df5425720f4b5f30335faed586bef129780146b7a2c"
  },
```
I mint a second a second asset:
```
tapcli assets mint --type normal \
                --name unlimitedRoylloCoin2 \
                --enable_emission true \
                --group_key 0349d60c6689bbbb2ffc9b30b45b96a3d6e5fd5fd01c867344c0054f6105048241
```
And as result:
```
"name": "unlimitedRoylloCoin2",
"asset_id": "8deae8ae25e1e46f8d74ba47b28860001965628d2c011f8f19f9e07dbecec7d0"
  "asset_group": {
    "raw_group_key": "02a97752df0a27efc44bf7bf0dcf7113b391ba6e1884cd5a78fe493f4da91e8f99",
    "tweaked_group_key": "02807f3a0009f9ed5c3ba0e3662c511619beb216e7f165b9ba77668c4835776b96",
    "asset_witness": "0140d5cccebf06fb9e26a8baa1fccf01f59c3c411a87dde8ff16e710fe0193eded77153dfb5d954f810b7f5920192f72f856b024e89f4eba1099087fb3dc4b3efe35"
  },
```
tweaked_group_key are different for both emissions.

### Expected behavior

It works if I remove ` --enable_emission true`. To void mistakes that could make people loose money, I think that you should not be allowed to use  `--group_key` AND `--enable_emission` at the same time.

**Comment by guggero** [comment]
Thanks for the issue.
I see how this might be confusing. Let me explain:
In general, the `--enable_emission` (btw., that is a boolean flag by itself and specifying `true` with a space might lead to it being seen as a positional argument, so better to avoid setting a value for boolean arguments) indicates that the issue minted with that command is _the very first in a group_. Any subsequent mint into that group should only use the `--group_anchor` (useful for minting into the same batch) or the `--group_key` (when referencing an asset from a batch that is already finalized or confirmed).

BUT: If you don't want to auto-generate a group key, you can also specify one when creating the very first asset of a group. For that you would also use the `--group_key` flag but in combination with `--enable_emission`.
But maybe we should use different flags for the two use cases? For example `--custom_group_key` when using with `--enable_emission` and `--anchor_group_key` when referencing an existing group?

**Comment by jharveyb** [comment]
I think this should have failed if enableEmission and a group key or anchor were set:

https://github.com/lightninglabs/taproot-assets/blob/e7859ee9e95896c4d5864149aee06970cedf160d/rpcserver.go#L345

So, `--enable_emission=true` for the group anchor, and then only `--group_key` or `--group_anchor` for following assets intended for the same group.

**Comment by dstadulis** [comment]
related to https://github.com/lightninglabs/taproot-assets/issues/236

**Comment by jharveyb** [comment]
Closed by #681 .
